src/common/interfaces/response.ts

```diff
@@ -0,0 +1,5 @@
+export interface IResult<T = Record<string, unknown>> {
```

Comment:
Take a look at this and let me know how it looks. Is this what we wanted?

Reply:
How about replacing `unknown` with `any`?

Reply:
> How about replacing `unknown` with `any`?

unknown is the type-safe counterpart of any. We needn't assign the result to anything. But, just return the data. So, unknown is the best bet here. But this is only used when we don't know the return type. Most of the times, we know the return type

Reply:
Alright. 
since we are specifying `T = Record<string, unknown>` passing lists as result will be possible?

Reply:
> Alright.
> since we are specifying `T = Record<string, unknown>` passing lists as result will be possible?

Nope. That's when we be like 
IResult<string[]> This way, we can

Reply:
Or do you think we should replace `T = Record<string, unknown>` with `T = Record<string, unknown> || Record<string, unknown>[]` or something to support both objects and list of objects?

Reply:
> Or do you think we should replace `T = Record<string, unknown>` with `T = Record<string, unknown> || Record<string, unknown>[]` or something to support both objects and list of objects?

That would make the flat lists complex.

Reply:
If we are `IResult<IData>` then all return types would be possible right? What will be implementation of IData?

Reply:
> If we are `IResult<IData>` then all return types would be possible right? What will be implementation of IData?

I didn't understand. Cosider this
```typescript
interface IResult<T = Record<string, unknown>> {
    status: 'OK' | 'NOT OK';
    error?: string;
    result: T | T[];
}
interface IData {
    username: string;
    password: string;
}

const obj: IResult<IData> = {
    status: 'OK',
    result: {
        username: 'sample',
        password: 'sample'
    }
}
const obj2: IResult<IData> = {
    status: 'OK',
    result: [{
        username: 'sample',
        password: 'sample'
    }]
}
console.log(obj);
console.log(obj2);
```

Reply:
Okay done, We'll go with this.